Uni merge/1.0.7

nenadV91 opened the pull request:
# NORMAL MERGE! DO NO SQUASH!
## Summary
- update to web3-react version 8 and all of the web3-react related packages
- disconnect from Metamask option
- general injected selector option if there is no Metamask wallet found
- unsupported network UI changes, now there is a network selector available
- pre-loading effect of tokens in the currency-select modal
- they also added Celo network and amplitude analytics but in this PR we are not using those

elena-zh reviewed (changes requested):
Hey @nenadV91 , great PR! I even thought that I would not find a lot of issues, but... 

- [ ] 1. It would be nice to make Toggles for the Recipient and Expert mode more visible, I the light mode they look  a bit better, but I'd apply changes for both light and dark modes
![toggles](https://user-images.githubusercontent.com/70885163/182663205-461b0496-be73-46f5-9303-369d225f1fe7.jpg)

- [ ] 2. Account: Add token to MM button is displayed to a not connected user/user that is connected to other wallets. 'View Contract' should be displayed for these cases
![add token - view contr](https://user-images.githubusercontent.com/70885163/182663663-5a5dfc28-3179-410b-973b-258dc31ebce7.jpg)

- [ ] 3. Connection modal: no 'New to markets' message in the new modal in comparison with the existing modal version
![no new message](https://user-images.githubusercontent.com/70885163/182663904-de991c51-16bb-4bc1-8a55-7f156ad9a095.jpg)

- [ ] 4. No error message when cancel connection to Fortmatic wallet
- [ ] 5. Connect to Fortmatic, then disconnect, refresh the page --> Fortmatic connection modal appears again and again
- [ ] 6. Connection modal is changed, and now it is not displayed to what wallet we are connecting. Leaving this point open to discussions as I'm not stick to it 
![image](https://user-images.githubusercontent.com/70885163/182664495-efb0138f-d740-4aa0-979b-f40c213d2720.png)

- [ ] 7. Again, unsupported network flow is changed, but I see some issues there: 
 - network selector looks differently than the main one
 - there is messages that we are showing to a user
 - 'select tokens' are displayed in the Swap field, no connect to wallet button
 
![messages](https://user-images.githubusercontent.com/70885163/182665529-9c7397af-e8b4-4a14-9a24-2f6dce517241.jpg)
![connect to unsupp network](https://user-images.githubusercontent.com/70885163/182665640-562c1cef-5336-4c34-bad2-6d46ba60f4a6.jpg)

- [ ] 8. I have got a crash of the app when I changed a network in the connected wallet to an unsupported one
![crash](https://user-images.githubusercontent.com/70885163/182665764-e31f69b8-99a1-48ac-8bcb-dc7fb54ee49d.jpg)

- [ ] 9. Network selector shows an unsupported network in the wallet (we used to not to show it, so I'm reporting it)
![polygon detected](https://user-images.githubusercontent.com/70885163/182666080-2c6ec1ab-c3d3-4377-9513-3ccce3c1a209.jpg)

- [ ] 10. When I connect a wallet with an unsupported network, I see 'Error' instead of the account modal. It opens empty Account modal along with the connection modal
![error calls](https://user-images.githubusercontent.com/70885163/182666285-02ae7748-a3d5-4e1e-8fcf-094a7f2d6d51.jpg)

- [ ] 11.Address is not visible when hover a mouse on it in the dark mode in the activity modal. In the light mode it becomes invisible when press on it. In both modes copy icon is missing near the address
![account](https://user-images.githubusercontent.com/70885163/182666534-9a9e56b9-81f6-41ff-a963-fe89a3fa53aa.jpg)

- [ ] 12. When I change a network to GC, I'm disconnected
- [ ] 13. When I add GC to MM, I see a warning there (need to write 'Gnosis' instead of the 'Gnosis chain' inside of the request to avoid this issue)
Besides, it is impossible to add GC network using network selector in a mobile device.
![gnosis](https://user-images.githubusercontent.com/70885163/182666803-5e3a154b-2f2c-4153-bf46-ab2956fba1f3.jpg)

- [ ] 14. Pending transaction tooltip does not appear when close Order Submitted modal (#783 PR)
- [ ] 15. I did not find 'Claim from locked GNO' section in the Account page when I connected to the account where it should be(for Rinkeby `0x57813f5C6e9636936161C33D8dfeAda8DDf49Fa0`)

- [ ] 16. There is no wallet name in the Activity modal when connect using WC
![no wallet name](https://user-images.githubusercontent.com/70885163/182667411-c1083b80-bddd-4d55-8f9c-893431edbf2f.jpg)

- [ ] 17. When I change a wallet and connect to it, 2 options are highlighted in the connection modal 
![2 options](https://user-images.githubusercontent.com/70885163/182667540-cc8ba825-e809-4bc0-bcaa-53c5765f4449.jpg)

- [ ] 18. When I disconnect MM, and refresh the page, MM is conencted again (due to it is not disconnected inside the MM wallet)

- [ ] 19. 'Disconnect' button is added to the app when open it in aa wallet integrated browser (Trust/imToken/1Inch/TokenPocket). However, it is impossible to disconnect a wallet there. Besides, a list of wallets to reconnect is added there (also, it is impossible to connect to these wallets when open the app inside another wallet's browser)
- [ ] 20. Network selector is enabled when connect to SC wallets (should not be displayed)

- [ ] 21. Balances are not loaded when connect to a wallet using WC in GC and Rinkeby.

elena-zh commented:
Hey @nenadV91 , I'm trying to reproduce the * case, and accidentally faced this issue: there is no 'pending' state on the account icon while the order is in 'Open' state
![image](https://user-images.githubusercontent.com/70885163/182887767-380cace2-b25c-4c2d-8234-a4df5a8790d9.png)

elena-zh commented:
@nenadV91 , 
**How to reproduce case 8:** 
1. Place an order
2. While the order is pending, switch to an unsupported network  --> the app is crashed.

alfetopito commented on src/components/analytics/GoogleAnalyticsProvider.tsx:
```diff
@@ -0,0 +1,51 @@
+/**
+ * Google Analytics Provider containing all methods used throughout app to log events to Google Analytics.
+ */
+export default class GoogleAnalyticsProvider {
```
We did something similar on our side, didn't we?

nenadV91 replied:
Yes, we have our own implementation

alfetopito reviewed (commented):
Reviewed 108/555 files

Continuing...

alfetopito commented on src/components/swap/TradePrice.tsx:
```diff
@@ -32,7 +32,12 @@ const StyledPriceContainer = styled.button`
-  const usdcPrice = useUSDCPrice(showInverted ? price.baseCurrency : price.quoteCurrency)
+  const usdcPrice = useStablecoinPrice(showInverted ? price.baseCurrency : price.quoteCurrency)
+  /*
+   * calculate needed amount of decimal prices, for prices between 0.95-1.05 use 4 decimal places
+   */
+  const p = Number(usdcPrice?.toFixed())
+  const visibleDecimalPlaces = p < 1.05 ? 4 : 2
```
Should we tell them it won't work as the comment describes?

nenadV91 replied:
https://github.com/Uniswap/interface/issues/4325 :+1:

alfetopito commented on .env:
```diff
@@ -77,8 +77,8 @@ REACT_APP_MOCK=true
 # IPFS uploading
-REACT_APP_PINATA_API_KEY=
-REACT_APP_PINATA_SECRET_API_KEY=
+REACT_APP_PINATA_API_KEY=foo
+REACT_APP_PINATA_SECRET_API_KEY=bar
```
NOOOOOO!!!

Remember to user your local `.env.local`

Then you'll never have this problem again :)

nenadV91 replied:
:rofl: So that is the solution, will use that one

alfetopito reviewed (commented):
Reviewed 144/557

Done with Uni original stuff, will start with `custom` folder now

alfetopito commented on src/custom/components/TransactionConfirmationModal/index.tsx:
```diff
@@ -152,12 +152,12 @@ const ButtonCustom = styled.button`
+//   width: 20px;
+//   max-height: 100%;
+//   margin: 0 10px 0 0;
+// `
```
This file is not a mod, is it?
I would be in favor of removing rather than commenting out

nenadV91 replied:
I'm not sure I follow. 
As far as I can see its a mod.

alfetopito replied:
I mean, the `*Mod` file in the same folder is a mod, this is the index file which is completely owned by us.
I'm in favor of removing stuff from this one rather than commenting out

nenadV91 replied:
Ok, removed comments

alfetopito commented on src/custom/components/WalletModal/InjectedOption.tsx:
```diff
@@ -0,0 +1,53 @@
+import { Trans } from '@lingui/macro'
```
Is there anything modded here?

nenadV91 replied:
It should be https://github.com/cowprotocol/cowswap/blob/uni-merge/1.0.7/src/custom/components/WalletModal/InjectedOption.tsx

alfetopito commented on src/custom/components/WalletModal/Option.tsx:
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
```
If no longer a mod, do we need this file?

nenadV91 replied:
Its still a mod that has some small css changes.

alfetopito commented on src/custom/components/WalletModal/WalletModalMod.tsx:
```diff
@@ -1,35 +1,35 @@
 // MOD imports
 import ModalMod from '@src/components/Modal'
+import { changeWalletAnalytics } from 'utils/analytics'
+import usePrevious from '@src/hooks/usePrevious'
```
@src import?

alfetopito commented on src/custom/components/Web3Status/Web3StatusMod.tsx:
```diff
@@ -139,43 +135,39 @@ function Sock() {
-export function Web3StatusInner({
-  pendingCount,
-  StatusIconComponent,
-  thereWasAProvider, //mod
```
Don't we still need this mod for something?

nenadV91 replied:
No, this is a part of previous eager connect to last wallet which is now changed

alfetopito commented on src/custom/constants/index.ts:
```diff
@@ -38,22 +37,6 @@ export const SAFE_APP_CODE = `${DEFAULT_APP_CODE}-SafeApp`
-  ...SUPPORTED_WALLETS_UNISWAP.WALLET_LINK,
-  mobile: true,
-}
-const DISABLED_WALLETS = /^(?:Portis|COINBASE_LINK)$/i
```
We might want to keep this

nenadV91 replied:
So this was basically only used here https://github.com/cowprotocol/cowswap/blob/619b1bec31b44fa741f97d7fb1bb00b549b47ce1/src/custom/constants/index.ts#L51 to filter the Uniswap defined connector options, but now, this is changed and the SUPPORTED_WALLETS const doesn't even exists and each connection option is defined separately in its own component so I think if they decide to add those wallets we will notice it

alfetopito reviewed (commented):
Reviewed 208/557

Done with components
Hooks next

alfetopito commented on src/custom/utils/switchToNetwork.ts:
```diff
@@ -1,87 +0,0 @@
-      // metamask's behavior when switching to the current network is just to return null (a no-op)
-      try {
-        await provider.request({
-          method: 'wallet_switchEthereumChain',
```
Is this logic captured somewhere else now?

nenadV91 replied:
Yes although its a bit different now https://github.com/cowprotocol/cowswap/blob/546840464cb7cddc97771b07f063ec2be3c60c0a/src/utils/switchChain.ts#L59, I think there is no request method on a provider anymore

alfetopito commented on src/hooks/Tokens.ts:
```diff
@@ -1,10 +1,11 @@
 import { useCurrencyFromMap, useTokenFromMapOrNetwork } from 'lib/hooks/useCurrency'
 import { getTokenFilter } from 'lib/hooks/useTokenList/filtering'
 import { useMemo } from 'react'
+import { isL2ChainId } from '@src/utils/chains'
```
path?

nenadV91 replied:
Has to be this one, because we use different chains

alfetopito commented on src/hooks/useAccountRiskCheck.ts:
```diff
@@ -0,0 +1,42 @@
+        const checkExpirationTime = storedTime ? parseInt(storedTime) : now - 1
+        if (checkExpirationTime < Date.now()) {
+          const headers = new Headers({ 'Content-Type': 'application/json' })
+          fetch('https://screening-worker.uniswap.workers.dev', {
```
What is this?

nenadV91 replied:
A new hook Uni added for checking of blocked accounts.
We are not using this for now

alfetopito reviewed (commented):
Reviewed 311/557

Now starting with original uni hooks

alfetopito commented on src/serviceWorker/document.ts:
```diff
@@ -0,0 +1,114 @@
+
+  // If this isn't app.uniswap.org (or a local build), skip.
+  // IPFS gateways may not have domain separation, so they cannot use document caching.
+  if (url.hostname !== 'app.uniswap.org' && !isDevelopment()) {
```
If we want to have service workers like they do, will have to mod this file

nenadV91 replied:
I will create a separate task for this.
It will probably take some time to understand this I suggest we do this after the merge at some point

alfetopito reviewed (commented):
557/557 reviewed \o/

alfetopito commented:
Behaviour wise, seems ok.
Connected Metamask, switched networks, placed and traded on rinkeby.
No issues that I could see at first glance

nenadV91 commented:
@elena-zh I've addressed a lot of issues you mentioned, please take a look and let me know what you find.
1. It would be nice to make Toggles for the Recipient and Expert mode more visible, I the light mode they look a bit better, but I'd apply changes for both light and dark modes
- fixed

 2. Account: Add token to MM button is displayed to a not connected user/user that is connected to other wallets. 'View Contract' should be displayed for these cases
- fixed

3. Connection modal: no 'New to markets' message in the new modal in comparison with the existing modal version
- uniswap removed this, do we wan't to keep it?

4. No error message when cancel connection to Fortmatic wallet
 - happens on Uniswap also

5. Connect to Fortmatic, then disconnect, refresh the page --> Fortmatic connection modal appears again and again
- fixed

 6. Connection modal is changed, and now it is not displayed to what wallet we are connecting. Leaving this point open to discussions as I'm not stick to it
- same as Uniswap

7. Again, unsupported network flow is changed, but I see some issues there:
- same as uniswap

8. I have got a crash of the app when I changed a network in the connected wallet to an unsupported one
- fixed

 9. Network selector shows an unsupported network in the wallet (we used to not to show it, so I'm reporting it)
- fixed

 10. When I connect a wallet with an unsupported network, I see 'Error' instead of the account modal. It opens empty Account modal along with the connection modal
- I'm not sure what the issue is here

11.Address is not visible when hover a mouse on it in the dark mode in the activity modal. In the light mode it becomes invisible when press on it. In both modes copy icon is missing near the address
- fixed

12. When I change a network to GC, I'm disconnected
- couldn't reproduce this

13. When I add GC to MM, I see a warning there (need to write 'Gnosis' instead of the 'Gnosis chain' inside of the request to avoid this issue)
- Doesn't this happen on prod also?

14. Pending transaction tooltip does not appear when close Order Submitted modal (#783 PR)
- fixed

 15. I did not find 'Claim from locked GNO' section in the Account page when I connected to the account where it should be(for Rinkeby 0x57813f5C6e9636936161C33D8dfeAda8DDf49Fa0)
- fixed

 16. There is no wallet name in the Activity modal when connect using WC
- fixed

 17. When I change a wallet and connect to it, 2 options are highlighted in the connection modal
- fixed

 18. When I disconnect MM, and refresh the page, MM is conencted again (due to it is not disconnected inside the MM wallet)
 - as far as I can see it happens only when you quickly refresh the page after the disconnect and that is because there isn't enough time for application to updated the localStorage with the new state, otherwise it works fine, so I don't think this is an actual problem.

  20. Network selector is enabled when connect to SC wallets (should not be displayed)
- fixed

21. Balances are not loaded when connect to a wallet using WC in GC and Rinkeby.
- fixed

elena-zh commented:
Hey @alfetopito , @anxolin , @fairlighteth , while testing the current PR changes I highlighted cases that are changed with the new Uniswap version (see quoted cases below)
All these issues are negotiable and it would be great to get your opinion whether we are leaving all these changes as they are or fix something to have some features like they are currently work in CowSwap. 

> * [ ]  3. Connection modal: no 'New to markets' message in the new modal in comparison with the existing modal version
>   ![no new message](https://user-images.githubusercontent.com/70885163/182663904-de991c51-16bb-4bc1-8a55-7f156ad9a095.jpg)
> * [ ]  4. No error message when cancel connection to Fortmatic wallet
> * [ ]  6. Connection modal is changed, and now it is not displayed to what wallet we are connecting. Leaving this point open to discussions as I'm not stick to it
>   ![image](https://user-images.githubusercontent.com/70885163/182664495-efb0138f-d740-4aa0-979b-f40c213d2720.png)
> * [ ]  7. Again, unsupported network flow is changed, but I see some issues there:
> * network selector looks differently than the main one
> * there is messages that we are showing to a user
> * 'select tokens' are displayed in the Swap field, no connect to wallet button
> 
> ![messages](https://user-images.githubusercontent.com/70885163/182665529-9c7397af-e8b4-4a14-9a24-2f6dce517241.jpg) ![connect to unsupp network](https://user-images.githubusercontent.com/70885163/182665640-562c1cef-5336-4c34-bad2-6d46ba60f4a6.jpg)
>

nenadV91 commented:
> Hey @nenadV91 , I'm trying to reproduce the * case, and accidentally faced this issue: there is no 'pending' state on the account icon while the order is in 'Open' state ![image](https://user-images.githubusercontent.com/70885163/182887767-380cace2-b25c-4c2d-8234-a4df5a8790d9.png)

Should be also fixed now

nenadV91 commented on .github/actions/setup/action.yml:
```diff
@@ -0,0 +1,21 @@
+name: Setup
+runs:
+  using: composite
```
Looks like a general setup action, but I've moved it to uniswap-original

nenadV91 commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,103 @@
+name: Test
```
Done

nenadV91 commented on .gitignore:
```diff
@@ -42,8 +37,6 @@ yarn-error.log*
 .idea/
 
-.vscode/
-
```
Restored

nenadV91 commented on .vscode/settings.json:
```diff
@@ -0,0 +1,19 @@
+  "[typescript]": {
+    "editor.defaultFormatter": "esbenp.prettier-vscode"
+  }
+}
```
Nope :smile:

nenadV91 commented on package.json:
```diff
@@ -85,29 +85,14 @@
-    "d3": "^7.0.0",
+    "@typescript-eslint/eslint-plugin": "^4",
+    "@typescript-eslint/parser": "^4",
+    "cypress": "^10.1.0",
```
Restored the prev version for now

nenadV91 commented on src/components/ConnectedAccountBlocked/index.tsx:
```diff
@@ -0,0 +1,58 @@
+  isOpen: boolean
+}
+
+export default function ConnectedAccountBlocked(props: ConnectedAccountBlockedProps) {
```
Nice catch. I've created a mod component for the TopLevelModals component that renders this component but for now this new component will be commented out and I've created a google doc for this.

nenadV91 commented on src/components/AccountDetails/Transaction.tsx:
```diff
@@ -1,4 +1,4 @@
-import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { useWeb3React } from '@web3-react/core'
```
Indeed :smile:

nenadV91 commented on src/components/Header/NetworkSelector.tsx:
```diff
@@ -1,22 +1,22 @@
 import { Trans } from '@lingui/macro'
-import { CHAIN_INFO } from 'constants/chainInfo'
+import { useWeb3React } from '@web3-react/core'
+import { getConnection } from '@src/connection/utils'
+import { getChainInfo } from '@src/constants/chainInfo'
```
Yea this is for Uniswap specific imports

nenadV91 commented on src/components/Header/NetworkSelector.tsx:
```diff
@@ -1,22 +1,22 @@
-
-import { useAppDispatch } from '../../state/hooks'
-import { switchToNetwork } from '../../utils/switchToNetwork'
+import { isChainAllowed, switchChain } from '@src/utils/switchChain'
```
Same as above, we have mods with different chains

nenadV91 commented on src/components/SearchModal/CurrencySearch.tsx:
```diff
@@ -1,29 +1,31 @@
 import Column from '../Column'
 import Row, { RowBetween, RowFixed } from '../Row'
-import CommonBases from 'components/SearchModal/CommonBases'
+import CommonBases from '@src/components/SearchModal/CommonBases'
```
Have to use this one

nenadV91 commented on src/components/TopLevelModals/index.tsx:
```diff
@@ -0,0 +1,23 @@
+    <>
+      <AddressClaimModal isOpen={addressClaimOpen} onDismiss={addressClaimToggle} />
+      <ConnectedAccountBlocked account={account} isOpen={open} />
+    </>
```
Done

nenadV91 commented on src/custom/components/AccountDetails/index.tsx:
```diff
@@ -133,21 +119,38 @@ export default function AccountDetails({
+
   const handleDisconnectClick = () => {
-    ;(connector as any).close()
-    localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
```
That whole implementation is removed.
Uniswap added their own implementation of this in `src/hooks/useEagerlyConnect.ts` and seems like the connectors now support `connectEagerly` out of the box.

nenadV91 commented on src/custom/hooks/useChangeNetworks.ts:
```diff
@@ -1,99 +0,0 @@
-type ChangeNetworksParams = Pick<ReturnType<typeof useActiveWeb3React>, 'account' | 'chainId' | 'library'>
-export type ChainSwitchCallbackOptions = { skipWalletToggle: boolean; skipToggle: boolean }
-
-export default function useChangeNetworks({ account, chainId, library }: ChangeNetworksParams) {
```
It seems that it was used only in NetworkSelector and Claim file which we don't use anymore. This had a lot of changes in this merge and in my opinion it is easier to keep it in sync with Uniswap merge for future merges which means removing it and keeping the logic in NetworkSelector

elena-zh commented:
Hey @nenadV91 , great changes!!! 

So here is the list of not fixed/new issues I have faced: 
1. Toggle recipient is not totally disabled when I am in the Expert mode. See the video: https://watch.screencastify.com/v/oTDUpVPy1X8I595JjNxd
2. Case 12: `When I change a network to GC, I'm disconnected`: actually, it appeared that it is not exactly related to the GC: I switched to it after several previous switches, so the issue was reproducible to me with it. So, the issues sounds this way: when I switch networks several time one after another, I can be disconnected in the end. see the video: https://watch.screencastify.com/v/cDxqMDOpqQxMSi8dr9wd
3. Connect to MM, then change wallet to WC. refresh the page --> you will be connected to MM instead of WC
4. Case 19 is not fixed: ` 19. 'Disconnect' button is added to the app when open it in aa wallet integrated browser (Trust/imToken/1Inch/TokenPocket). However, it is impossible to disconnect a wallet there. Besides, a list of wallets to reconnect is added there (also, it is impossible to connect to these wallets when open the app inside another wallet's browser)`
5. When I changed a network in MM being connected to MM and WC, network selector disappeared (see the end of the video: https://watch.screencastify.com/v/TRiEg1sMgdAM1l9oeagH )  (might be an edge case)

Thank you!

nenadV91 commented on src/custom/hooks/web3.ts:
```diff
@@ -1,230 +0,0 @@
-  }, [connectInjected, active, connectSafe, triedSafe, reconnectUninjectedProvider]) // intentionally only running on mount (make sure it's only mounted once :))
-
-  // wait until we get confirmation of a connection to flip the flag
-  useEffect(() => {
```
I've mentioned this but now this is implemented differently

nenadV91 commented on src/custom/pages/Account/Balances.tsx:
```diff
@@ -55,6 +56,8 @@ export default function Profile() {
   const setSwapVCowStatus = useSetSwapVCowStatus()
   const swapVCowStatus = useSwapVCowStatus()
 
+  const isMetaMask = (connector as MetaMask)?.provider?.isMetaMask
```
That is what I thought, but its not what it looks like.

`export function getIsMetaMask(): boolean {
  return window.ethereum?.isMetaMask ?? false
}`

It actually just checks if there is a Metamask wallet and not if the current connected wallet is metamask.
I think I've made some mistakes before where I've used that method instead of this so I will have to check that again.

nenadV91 commented on src/custom/pages/Account/Balances.tsx:
```diff
@@ -76,15 +79,15 @@ export default function Profile() {
   )
 
   const isCardsLoading = useMemo(() => {
-    let output = isVCowLoading || isLockedGnoLoading || !library
```
Nice catch, restored

nenadV91 commented on src/custom/state/user/actions.ts:
```diff
@@ -1,6 +0,0 @@
-export * from '@src/state/user/actions'
-
-export const toggleFavouriteToken = createAction<{ serializedToken: SerializedToken }>('user/toggleFavouriteToken')
-export const removeAllFavouriteTokens = createAction<{ chainId: number }>('user/removeAllFavouriteTokens')
```
The user reducer is changed to use slice so the actions are now being exported from the reducer file https://github.com/cowprotocol/cowswap/blob/a4d1be1cefdd6cbf915c9e4918303573e4499043/src/custom/state/user/reducer.ts#L232

nenadV91 commented:
@elena-zh Regarding your latest comment

1 Toggle recipient is not totally disabled when I am in the Expert mode. See the video: https://watch.screencastify.com/v/oTDUpVPy1X8I595JjNxd
- Fixed

2 Case 12: When I change a network to GC, I'm disconnected
- Couldn't reproduce it

3 Connect to MM, then change wallet to WC. refresh the page --> you will be connected to MM instead of WC
- Couldn't reproduce it, works fine for me

4 Case 19 is not fixed: 19. 'Disconnect' button...
- Should be fixed now

5 When I changed a network in MM being connected to MM and WC...
- Couldn't reproduce, I've noticed that in that video the Metamask connection lagged and then it re-connected to wallet-connect again

nenadV91 commented on src/custom/pages/Account/LockedGnoVesting/claimData/index.ts:
```diff
@@ -3,6 +3,7 @@ import mainnetIndex from './mainnet.json'
 import rinkebyIndex from './rinkeby.json'
 import gnosisChainIndex from './gnosisChain.json'
 import goerliIndex from './goerli.json'
+import { supportedChainId } from '@src/custom/utils/supportedChainId'
```
Fixed

nenadV91 commented on cypress.config.ts:
```diff
@@ -0,0 +1,20 @@
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  projectId: 'yp82ef',
```
I've downgraded the cypress for now and we don't need this file

elena-zh commented:
@nenadV91 , the PR looks nicer and nicer. 
And again. 
1. 
> > 2 Case 12: When I change a network to GC, I'm disconnected
> 
> Couldn't reproduce it
 I can reproduce it. here the console log that I see when I change a network and become disconnected. 
![image](https://user-images.githubusercontent.com/70885163/183699826-a188bc9f-d2e2-4fb8-812c-9e8017249d19.png)

2. Network selector still looks different when connected to an unsupported network. Despite of the fact that 'unsupported network' flow is still under discussion, I think that network selector should look the same all the time
![looks different](https://user-images.githubusercontent.com/70885163/183700646-e00bf415-5d5b-4d71-b378-fb566744fb8e.jpg)
3. When I press on the 'Open in Coinbase app' button when try to connect the app in a mobile view, I'm banned, and navigated to the uniswap page
See the video: 

https://user-images.githubusercontent.com/70885163/183701042-b8c6929e-09f8-4a5c-b910-e82f05df442c.mp4

4. Token lists are jumping when I switch them on/off. Looks like a dancing hell. I have to notice that token lists are not fixed on Prod now as well, but I did notice that they may jump this way there. See the video: https://watch.screencastify.com/v/hpjkqeWkLRqOCgOHyi3U

Thanks!

nenadV91 commented:
@elena-zh 
1. Still can't reproduce it
2. I would have to disagree with this one. I think this is a solid solution from Uniswap, but that is my opinion
3. Fixed
4. Fixed

elena-zh commented:
Hey @nenadV91 
> 1. Still can't reproduce it

I can. I have been able to reproduce it 2 times today. Try switching between networks using network selector. See the video: https://watch.screencastify.com/v/73dz7qpRc1K6pe1hLZlF

2. Yes, leaving this as it is (according to opinions in https://docs.google.com/document/d/1V7rUrgcz59YiW9eVzncY7TRx8nGU6TT_DWNU-jB29Pc/edit?usp=sharing)
3. Now the link navigates me to the Coinbase app, but I can't be connected there. Could we please navigate to the Coinbase integrated browser and load the app there (you can check how it works in Prod version now)
4. LGTM!

Thanks!

nenadV91 commented on .github/uniswap-original/workflows/check-pr-title.yaml:
```diff
@@ -8,8 +8,8 @@ on:
-  check-pr-title:
-    name: Check PR Title
+  # Ensures that the PR title adheres to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+  conventional-commit:
```
I've created a separate task for this.
Lets look into this after the merge

nenadV91 commented on src/custom/components/Header/NetworkSelector/NetworkSelectorMod.tsx:
```diff
@@ -263,9 +286,11 @@ function Row({
+            <ExternalLink href={getExplorerBaseUrl(chainId)}>
+              <Trans>CoW Protocol Explorer</Trans> <LinkOutCircle />
+            </ExternalLink>
+          ) : null}
```
Done

elena-zh commented:
> @elena-zh But isn't the `Coinbase integrated browser` in the Coinbase app? As far as I can see it works the same as on prod.

@nenadV91  , I have retested it again and it works fine to me now. 
When I wrote you my previous comment, I was navigated to the Swap page in the Coinbase app (maybe glitch or so). Sorry about misleading you.

nenadV91 commented:
@elena-zh No worries, could be related to cached files. 
Anyways, good to hear it works now.

elena-zh commented:
Then, the issue with MM disconnect is left. But I can create a separate task for this. Let me know please.

elena-zh reviewed (approved):
Hey @nenadV91 , 
given the fact that I will review the PR before the release, and that it is actually looks good (you have done a great job!), I'm approving it. 
I have created 2 low-prio issues related to connection: #923 and #925 that are still here. 
Also , it would be nice to have the same list of supported networks on the pop-up and the modal when a user is connected to an unsupported network: 
![image](https://user-images.githubusercontent.com/70885163/184654271-865f12ff-c0ea-41d4-9230-8f02683c0d15.png)

Plus it would be great to add a colon after 'networks' in the Activity modal. 

Thanks!

elena-zh commented:
Hey @nenadV91 , I have found an issue that the app is crashed when remove a token list: 
![image](https://user-images.githubusercontent.com/70885163/184865035-3b5f4daf-f981-4819-99a4-de83043388bf.png)

Could you please take a look at it?

nenadV91 commented:
@elena-zh The popup issue will be fixed in one of the following PR's related to the account page.

elena-zh commented:
I have verified the issue with crash on a toke list removal action, and it is no longer reproducible. 
But I have found another UI issue related to token list: token list source address exceeds the import modal area. 

**To reproduce:**

1. Open token lists in Gorli, press on Manage token lists
2. Remove GP Goerli token list
3. Then open token lists, press on Manage token lists
4. Paste this source https://raw.githubusercontent.com/cowprotocol/cowswap/main/src/custom/tokens/goerli-token-list.json
5. Press on Import

![image](https://user-images.githubusercontent.com/70885163/185085618-48a454ee-d4f6-41f0-96db-41269c3f51c6.png)

elena-zh commented:
@nenadV91 , I noticed that the last issue is reproducible on Prod, so I opened a separate task for this #938
